July meeting notes (#337)
tc39/tc39-notes: meetings/2024-07/july-30.md

PFC: Okay. I recognize that opinion as well.

CDA: There’s a reply from SYG.

SYG: I want to see if DLM can give more colour to why is against it? Is it like the principle of changes to Stage 3 things? Is it something in practice with the SpiderMonkey implementation?

DLM: It’s just the principle. ABL has been doing a huge amount of work to get this ready. If we were to do it, it would be a small change. But it’s a small change on the steady series of small or larger series over the course of years. But it’s not an implementation concern. It’s more a principal concern.

SYG: Thanks

> DE: I apologize for muddying the waters a bit by voicing support for this in the chat. When the coercion topic came up, this is one of my original concerns that we not apply it to Stage 3 proposals. If it were possible to do this very cheaply, I think it would be nice. But I guess the—I would really want to conclude today on in principle whether we’re making this change. If we don’t conclude today we’re making this change, we’re not doing it for Temporal because as pFC said, it’s already shippable. The work that PFC did that went into the removal was kind of more significant than initially expected. I think he could tell it was going to be a lot of work. But there are just a lot of places where coercion happens. It wouldn’t be realistic.

Reviewer: ```suggestion
DE: I apologize for muddying the waters a bit by voicing support for removing coercion in Temporal in the Matrix chat. When the coercion topic came up, this is one of my original concerns that we not apply it to Stage 3 proposals. If it were possible to do this very cheaply, I think it would be nice. But I guess the—I would really want to conclude today on in principle whether we’re making this change. If we don’t conclude today we’re making this change, we’re not doing it for Temporal because as PFC said, it’s already shippable. The work that PFC did that went into the removal was kind of more significant than initially expected. I think he could tell it was going to be a lot of work. But there are just a lot of places where coercion happens. It wouldn’t be realistic.
```

WH: I just wanted to express a preference for `zipKeyed`.

ACE: I think that looks like the queue. MF, are you happy with—do you want to ask if we can consensus for changing to zipKeyed?

MF: Sure, yeah. Can we rename zipToObject to zipKeyed for joint iteration?

MM: I want to explicitly note that if we do this, we should likely follow the parallelism suggested by this presentation for promise.all. Not that we need to vote on that. But we should keep in mind that’s probably what we’re going to do if we do it here. And on those grounds. I have no objection.

MF: I agree.

MM: Okay. You have some support. Chris Cole says I like `iterator.zip` and iterator.zipKeyed. CM says +1 to zipkeyed.

CDA: Any objections?

> CDA: You have another +1 for zipkeyed and different keys from JSON. Now that I’m reading it there’s no end of message there. Did you want to speak?

Reviewer 2: ```suggestion
CDA: You have another +1 for zipkeyed from JWS. Now that I’m reading it there’s no end of message there. Did you want to speak?
```

PFC: I believe this already went to Stage 3.

DE Try to track down the meeting where that happened and I will update the proposals list.

PFC: I’m looking through the old agendas right now.

PKA: Okay. Let’s go on to the next topic.

CDA: It’s very much active, so I think we can –

PKA: Great.

PKA: Glad to get that updated. Next we have symbol predicates and I believe both the champions are here. Can anybody make a comment?

CDA: This is from JHD?

JHD: Yeah, it’s still—it’s both still active and also blocked. I have to—there are two predicates and one of them is not obstructed and the other one I need to come up with a more compelling case for to convince V8 in particular although there may be other engines that have concerns, so yeah.

PKA: Great, thanks for the update. Next we have module declarations.

NRO: This is like not recently because we have a lot of modules proposals going on. So we’re just like waiting for the one that had to stabilize before continuing with this one and m module expressions.

DE: I think the plan is we will pick this up again, right?

NRO: Yes.

> DE: Some interesting feedback from the committee would be if people feel like the way that we were previously going of using this kind of lexical name space but statically was acceptable. So if you have opinions about that, either shut them out now or come to some of the modules calls that we have on the TC39 calendar. I would like to go the way that was previously proposed where we—which NRO has fully written out a good specification for where it’s modules are lexically scoped variables but they’re available kind of at static semantics time the way that, you know, your let declarations are checked for duplicates.

Reviewer: ```suggestion
DE: Some interesting feedback from the committee would be if people feel like the way that we were previously going of using this kind of lexical name space but statically was acceptable. So if you have opinions about that, either shout them out now or come to some of the modules calls that we have on the TC39 calendar. I would like to go the way that was previously proposed where we—which NRO has fully written out a good specification for where it’s modules are lexically scoped variables but they’re available kind of at static semantics time the way that, you know, your let declarations are checked for duplicates.
```

SYG: So a question, is this up cert.

JHD: Yes it is. But renamed.

DE: The names are kind of silly but I think this is a widely requested feature. There is a project i think in the planning stage with the University of Bergen students getting involves here, so it would be very useful to have any committee feedback on this organized for them so they can take up the proposal.. Any opinions or any cross reference to any opinions would be helpful?

CDA: MF has a plus one and the one thing that we are out a Champion. And so would anyone like to take up the mantle?

DE: I just mentioned there is a group that intends to take this on. Um, if you want to work with that group, then I can get you in touch.

CDA: Okay.

SYG: That is a bit cryptic.

DE: I said the University of Bergen is trying to take this up.

CDA: Oh okay.

DE: So, yeah.

CDA: Okay I misunderstood.

> DE: I’ll fix it in the notes.

Reviewer: ```suggestion
DE: The names are kind of silly but I think this is a widely requested feature. There is a project i think in the planning stage with the University of Bergen students getting involves here, so it would be very useful to have any committee feedback on this organized for them so they can take up the proposal.. Any opinions or any cross reference to any opinions would be helpful?

[There was confusion on whether the University of Bergen students would champion this proposal; DLM later clarified that he would work with them on championing it.]
```

PKA: So dynamic import host adjustment, and so the Champion is KOT, not known to be active in the committee, if if anybody knows about this proposal, please share?

NRO: This proposal has been transformed to a request and has had some concerns for a while. I think i might be—yes I was mistaken. I sample working with host types for a while but there is no updates on this proposal and not has been brought up with them.

> DE: Okay the author of this is, you know, in contact with a number of people in TC39 who are working on trusted types. So can we give some one the action to get in touch and see what everyone wants to do with this?

Reviewer: ```suggestion
DE: Okay the author of this is, you know, in contact with a number of people in TC39 who are working on Trusted Types. Can we give someone the action to get in touch and see what everyone wants to do with this?
```

NRO: I will attend to that.

DE: Thank you for much Nicolo.

PKA: Okay collection normalization? Anyone has status on this proposal or have any interest on taking it out?

> DE: ACE actually brought up this design space in his recreation and tuples and it is linked but it can be done separately.

Reviewer: ```suggestion
DE: ACE actually brought up this design space in his Records and Tuples presentation. It is linked but it can be done separately.
```

DE: JHD, What was the compromise?

JHD: So, my understanding of the—my recollection is that the prime mare objection was the desire to treat a map and set agnostically so I can have a design function and does not care if it is a map or a set. If I remember the correct proposal and the compromise was that if you provide—that you can provide one of the two—I will have to refresh my memory but something that would allow for that use case, but in the majority case, it would also allow for—it would allow for both mental models that is a set has only values or a set has only keys as well that I want to be agnostic of map versus set and I can dig up more content in the future. But essentially that was the compromise as I recall.

DE: Okay, great. Do you want to see if anyone wants to be a coChampion on you on this?

JHD: I will come back on the next plenary after I have done some research because I need to be confident about how much work is involved and how much room I have on my plate before even Championing it. But I will leave it through until then at least.

DE: Okay sounds good.

PKA: Okay thanks I will circle back to JRL?

> JRL: Perfect, so I plugged in the headphones and it did not switch back to the proper mic and so intrinsic dedents and pay path length is no longer a member and just member and there is no current blockers, I just have not written the test 262 test to get this 2.7.

Reviewer: ```suggestion
JRL: About `String.dedent`: It was championed by PayPal, who is is no longer a member. There are no current blockers, I just have not written the test 262 test to get this 2.7.
```

JHD: I would caution trying to built implementation of things that are not at Stage 2.7 because that the change significantly.

DE: The group is aware of this and many are coming to TC39 as well. And is because you have corrections? Or?

DLM: Sorry, I wanted to say that I would be one of mentors for this and the implementation in SpiderMonkey

AKI: MBH said that this is going to be a learning exercise for students and trying to bring them 2.7, and DLM, who was just speaking said they will be the mentor for SpiderMonkey for that implementation as part of process of student’s learning on how to put something into an engine? Is that clear?

DLM: Good from my point.

CDA: Okay that is great and I am happy to hear that is happening. And so not necessarily spend a bunch of time on this. But we would still be great to have a Champion for this proposal.

DE: Yes it was—okay go ahead. Because it was already clarified that you know this group of people is going to be working on bringing it to committee.

PKA: So our final item is destructuring private fields.

JRL: I am no longer working on this although DE has a proposal that would assume it entirely.

> DE: I have a proposal that would not subsume it but make the region destructure private fields okay. This is something I have written en—I don’t have it written up in a clear way I mean, but for the old topic of private name declaration and I am wanting if we can use the syntax hash paren and expression. And instead of hash ex. This would then free up the destructure private fields syntax to what JRL proposed before and I would like to come back and present that sketch. I don’t know if I will have energy to Champion myself but given that, the destructure private fields proposal, given sort of consensus and principle at this, we can proceed with what JRL was proposing before. Do you have thoughts on this?

Reviewer: ```suggestion
DE: I have a proposal for how we could handle private name declarations, and this proposal would make it clear that the current destructuring private fields proposal is OK. It makes it clear that the syntax space used in destructuring private fields won't clash with something else. This would then free up the destructure private fields syntax to what JRL proposed before. I would like to come back and present that sketch of this alternative feature. I don’t know if I will have energy to champion the proposal myself, but if folks are convinced about the direction, then we can proceed with what JRL was proposing before. Do you have thoughts on this?
```

NRO: It is its own on proposal.

SFC: Um, yeah this came up originally with the fluent iteration which this is not of locales but a lot of the other entities that exists such as currencies, and calendar steps and measurable p unit and things like that but we don’t—I think closest that we have to supportive locales is that the supportive locale of on function and that will give a list and force you out of that and in principle and you can have supportedLocaleOf with all possible locale and what support of that locales of that function will call.

DE: Right, so I am confused by this presentation because I thought all browsers adopted the strategy of having a fix set of locales that are there. That is invariant across what you—so you want it upped.

BAN: That is current behavior, and what we like it do is that browser can have dynamic locale and locale data so long As it does not reveal servers that has been installed. And something for the future.

DE: All the API’s async, what is API are you suggesting would allow this?

BAN: I am not suggesting an API and simply noting if you allow this and if you allow dynamic demonstration of locale data, you do not locate by whatever means.

> DE: Okay thanks for it is context.

Reviewer: ```suggestion
DE: Okay thanks for the context.
```
